src/core/metrics/app-metrics.ts

```diff
@@ -177,27 +178,26 @@ export class AppMetrics implements Metric {
 
       if (Object.values(this.watchList).every(Boolean)) {
         // watchList completed
-
-        // we need to wait (again) EVENT_MAP_IDLE after all events from watchList
-        // to determite that app is ready
         if (name !== EVENT_MAP_IDLE) {
           if (KONTUR_METRICS_DEBUG) {
-            console.warn('metrics waiting final mapIdle', this.watchList);
+            console.warn('metrics waitForMapFullyRendered', this.watchList);
           }
-
-          this.watchList[EVENT_MAP_IDLE] = null;
-
-          // repaint to ensure EVENT_MAP_IDLE will be triggered
-          // FIXME: replace globalThis?.KONTUR_MAP after init refactor
-          globalThis?.KONTUR_MAP.triggerRepaint();
-        } else {
-          // app is ready
-          this.cleanup();
-          this.report('ready', timing);
-          this.sendReports();
-
-          emitReadyForScreenshot(globalThis?.KONTUR_MAP);
-          return;
+          waitForMapFullyRendered(globalThis?.KONTUR_MAP, { timeout: 30000 })
+            .then(() => {
+              const timing = performance.now();
+              this.report('ready', timing);
+              const eventReadyEvent = new Event('event_ready_for_screenshot');
+              globalThis.dispatchEvent(eventReadyEvent);
+            })
+            .catch((error) => {
+              console.error('Map render detection failed:', error);
```

Comment:
Do we send any metrics event when the map times out?

Reply:
times out issues will be visible in BE monitoring as its BE fails initially, here we have 30s timelimit which is enough to see how's FE doing